### DISCUSSION TITLE
How to enable SSL termination at the application level, not NGINX

### DISCUSSION THREAD

**Discussion by macadev** [dokku/dokku#4877] [Q&A] [answered]
Hi Dokku maintainers, thanks for your work on this awesome project.

### Description of problem

I have been trying to see if there's an easy way to do the following: host a node.js application using Dokku and NGINX but have the node.js application handle SSL. This would mean that NGINX would simply let HTTPS traffic passthrough to the node.js application, rather than do the decryption / encryption itself.

The reason I want to do this is to evaluate whether this improves the performance of my application in terms of CPU usage. My goal is to get the most out of the cheapest digital ocean boxes.

### What I have tried

I have faced the following problems trying to create this setup:

1. The Dokku let's encrypt plugin installs the certs outside of the application container. For the Node.js app to handle SSL I would need to have the certs copied into the container
2. I've tinkered with the NGINX configuration to simply passthrough HTTPS traffic to the application container on port 443, but this has proved quite difficult due to my lack of experience with NGINX.

### Ask

Does anybody know of any example applications that do what I mention above?

Thanks for your time!

**Comment by josegonzalez** [comment, marked as answer]
I don't think this will be possible unless you use nginx as a [tcp passthru proxy](https://stackoverflow.com/a/46413044/1515875), which then means each of your individual processes will be processing ssl. This is definitely not something I'd support in  the core, as I think if I did, folks would probably misconfigure things more often than not.

I don't think that will actually improve throughput, as I'd trust nginx to have better optimized ssl termination than nodejs libraries. Thats not to mention that one probably specifically has folks go through and figure out security issues, where the other appears to have fun userland issues crop up every so often.
